### PR TITLE
feat: announce waterfall eol

### PR DESCRIPTION
**Pull request by olijeffers0n**

### CODE REVIEW

File: docs/waterfall/README.mdx [hunk 1]
```diff
@@ -1,3 +1,8 @@
+---
+sunsetted: true
+sunsettingMessage: We recommend you transition to Velocity.
```
Comment (notTamion) [inline docs/waterfall/README.mdx:3]: Make "Velocity" link to velocity docs

**Review by zlataovce — APPROVED** [review]

**Comment by MiniDigger** [comment]
do we really need to copy in the whole theme when we (at least from a quick glance) only modify one file?

**Comment by olijeffers0n** [comment]
> do we really need to copy in the whole theme when we (at least from a quick glance) only modify one file?

Unfortunately when you eject it includes all those components. I tried deleting some and had some issues however I believe that may have been caused by something else I have now fixed, so I’ll have a lool

File: src/theme/DocItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            <ReactMarkdown className={style.reactMarkDown} >{"**" + message + "**"}</ReactMarkdown>
+        </div>
+    );
+}
+
+export default function DocItem(props: Props): JSX.Element {
+    const docHtmlClassName = `docs-doc-id-${props.content.metadata.id}`;
+    const MDXComponent = props.content;
+    const sunsettedPage = props.content?.frontMatter?.sunsetted === true;
+    const sunsettingMessage = "This project has been sunsetted and is no longer maintained. " + props.content?.frontMatter?.sunsettingMessage || "";
```
Comment (natereprogle) [inline src/theme/DocItem/index.tsx:22]: Sunsetted is past tense and past participle. Sunsetted is also only grammatically correct when it is a noun or adjective. In this case, the word “sunset” is acting as a verb, meaning it doesn’t need “-ed”. Therefore, “sunsetted” is incorrect and “sunset” needs to be used here.

**Review by natereprogle — CHANGES_REQUESTED** [review]

File: src/theme/DocItem/index.tsx (hunk 2 above)
Comment (natereprogle) [reply src/theme/DocItem/index.tsx:22]: Btw, **well aware** this is one of _those_ 🤓 reviews, just one to try and get credit ([Similar to Beg Bounties](https://www.troyhunt.com/beg-bounties/)), however with how large the project is and how many eyes are put on it each month, maintaining professional and correct customer-facing info is important. I genuinely couldn't care less if this is fixed or not, just thought I'd point it out. Regardless, I'm glad to see Waterfall being sunset, Velocity really is vastly superior and I'm glad this project has reached EOL. Thanks for all you do for the community! 😄

Comment (zml2008) [reply src/theme/DocItem/index.tsx:22]: "customer" 😆

Comment (natereprogle) [reply src/theme/DocItem/index.tsx:22]: > "customer" 😆

What else do you want to call it? 💀

File: src/css/custom.css [hunk 3]
```diff
@@ -31,6 +31,20 @@ html[data-theme="dark"] {
...
+  width: 100%;
+  min-height: 5rem;
+  margin: 10px 0 20px 0;
+  padding: 15px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 8px;
+  border-left: 5px solid #fa383e;
+  background-color: rgb(75, 17, 19);
```
Comment (natereprogle) [inline src/css/custom.css:44]: image
Has this CSS been validated for light mode? It appears this way on Firefox and Chromium based browsers.

File: docs/waterfall/README.mdx [hunk 4]
```diff
@@ -1,18 +1,15 @@
+---
+eol: true
+eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information see the [announcement](https://forums.papermc.io/threads/1088/).
```
Comment (zlataovce) [inline docs/waterfall/README.mdx:3]:
```suggestion
eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information, see the [announcement](https://forums.papermc.io/threads/1088/).
```

File: docs/waterfall/configuration.mdx [hunk 5]
```diff
@@ -1,6 +1,8 @@
 ---
 slug: /configuration
 description: Configuration guide for Waterfall.
+eol: true
+eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information see the [announcement](https://forums.papermc.io/threads/1088/).
```
Comment (zlataovce) [inline docs/waterfall/configuration.mdx:5]:
```suggestion
eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information, see the [announcement](https://forums.papermc.io/threads/1088/).
```

File: docs/waterfall/getting-started.md [hunk 6]
```diff
@@ -1,6 +1,8 @@
 ---
 slug: /getting-started
 description: How to get started with running Waterfall.
+eol: true
+eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information see the [announcement](https://forums.papermc.io/threads/1088/).
```
Comment (zlataovce) [inline docs/waterfall/getting-started.md:5]:
```suggestion
eolMessage: We recommend you transition to [Velocity](https://papermc.io/software/velocity). For more information, see the [announcement](https://forums.papermc.io/threads/1088/).
```

File: src/theme/DocItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+            <ReactMarkdown className={style.reactMarkDown} >{"**" + message + "**"}</ReactMarkdown>
+        </div>
+    );
+}
+
+export default function DocItem(props: Props): JSX.Element {
+    const docHtmlClassName = `docs-doc-id-${props.content.metadata.id}`;
+    const MDXComponent = props.content;
+    const eolPage = props.content?.frontMatter?.eol === true;
+    const eolMessage = "This project has reached and of life and is no longer maintained. " + props.content?.frontMatter?.eolMessage || "";
```
Comment (zlataovce) [inline src/theme/DocItem/index.tsx:22]:
```suggestion
    const eolMessage = "This project has reached end of life and is no longer maintained. " + props.content?.frontMatter?.eolMessage || "";
```

File: src/theme/DocItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import {HtmlClassNameProvider} from '@docusaurus/theme-common';
+import {DocProvider} from '@docusaurus/theme-common/internal';
+import DocItemMetadata from '@theme/DocItem/Metadata';
+import DocItemLayout from '@theme/DocItem/Layout';
+import ReactMarkdown from 'react-markdown';
+import style from '@site/src/css/markdown-styles.module.css';
+import type {Props} from '@theme/DocItem';
+
+function EOLPage(message: string) {
```
Comment (zlataovce) [inline src/theme/DocItem/index.tsx:10]: function name seems a little odd, considering it's not a page
```suggestion
function EOLMessage(message: string): JSX.Element {
```

File: src/theme/DocItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            <ReactMarkdown className={style.reactMarkDown} >{"**" + message + "**"}</ReactMarkdown>
+        </div>
+    );
+}
+
+export default function DocItem(props: Props): JSX.Element {
+    const docHtmlClassName = `docs-doc-id-${props.content.metadata.id}`;
+    const MDXComponent = props.content;
+    const eolPage = props.content?.frontMatter?.eol === true;
+    const eolMessage = "This project has reached end of life and is no longer maintained. " + props.content?.frontMatter?.eolMessage || "";
```
Comment (zlataovce) [inline src/theme/DocItem/index.tsx:22]: there could be an explicit line break before the message, I think it looks a little better:

image

**Review by zlataovce — COMMENTED** [review]
`eolMessage` should probably be renamed to `eol_message` to stay consistent with [other front matter properties](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-content-docs#markdown-front-matter)

**Pull request merged**